sds_data_manager/constructs/ialirt_compute_resources.py

```diff
@@ -0,0 +1,102 @@
+"""
+This module provides the IalirtEC2Resources class which sets up I-Alirt
+compute resources utilizing EC2 as the compute environment (for now).
+"""
+from aws_cdk import aws_ec2 as ec2
+from aws_cdk import aws_ecr as ecr
+from aws_cdk import aws_iam as iam
+from constructs import Construct
+
+
+class IalirtEC2Resources(Construct):
```

Comment:
I was in favor of construct before but in this case, I feel like this could be turned into Stack like others since it only creates EC2 stuffs. construct is helpful if things are very custom.